### PR TITLE
fix: urlParser will incorrect parsing url like http://abc.com/xxx@xxx/a/b

### PR DESCRIPTION
**Pull request by wangzhen2271**
Fixes  issue #1509

## Changes

urlParser incorrect url which "@" not in first segment, like "http://abc.com/xxx@xxx".
So I modified that it will judge if "@" is in first segment, that means before the first "/",  otherwise ignore handle it.

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: ext/include/opentelemetry/ext/http/common/url_parser.h [hunk 1]
```diff
@@ -52,11 +52,17 @@ class UrlParser
...
+    size_t pos2 = url_.find_first_of("/", cpos);
+    if (pos1 != std::string::npos)
     {
       // TODO - handle credentials
-      cpos = pos + 1;
+      if (pos2 == std::string::npos || pos1 < pos2)
+      {
+        pos = pos1;
+        cpos = pos + 1;
+      }
```
Comment (lalitb) [inline ext/include/opentelemetry/ext/http/common/url_parser.h:64]: Nit
```cpp
        pos = pos1;
        cpos = pos + 1;
```

to

```
   cpos = pos1 + 1;
```

**Review by lalitb — APPROVED** [review]
Thanks for fix. You need to fix the code format as described here - https://github.com/open-telemetry/opentelemetry-cpp/blob/main/CONTRIBUTING.md#how-to-send-pull-requests

File: ext/include/opentelemetry/ext/http/common/url_parser.h [hunk 2]
```diff
@@ -52,11 +52,17 @@ class UrlParser
...
+
```
Comment (ThomsonTan) [inline ext/include/opentelemetry/ext/http/common/url_parser.h:65]: nit: remove this empty line.

**Review by ThomsonTan — APPROVED** [review]

**Review by owent — APPROVED** [review]
LGTM

**Pull request merged**